library/src/main/scala/scala/scalajs/js/Object.scala

```diff
@@ -202,6 +220,42 @@ object Object extends Object {
    */
   def preventExtensions(o: Object): o.type = native
 
+  /** <span class="badge badge-ecma2015" style="float: right;">ECMAScript 2015</span>
+   *
+   *  Object.is() determines whether two values are the same value. Two values
+   *  are the same if one of the following holds:
+   *
+   *  <ul>
+   *    <li>both undefined
+   *    <li>both null
+   *    <li>both true or both false
+   *    <li>both strings of the same length with the same characters in the same
+   *        order
+   *    <li>both the same object (means both object have same reference)
+   *    <li>both numbers and
+   *      <ul>
+   *        <li>both +0
+   *        <li>both -0
+   *        <li>both NaN
+   *        <li>or both non-zero and both not NaN and both have the same value
+   *      </ul>
+   *    </li>
+   *  </ul>
+   *
+   * This is not the same as being equal according to the `==` operator. The
+   * `==`` operator applies various coercions to both sides (if they are not the
+   * same Type) before testing for equality (resulting in such behavior as
+   * `"" == false`` being `true`), but Object.is doesn't coerce either value.
+   *
+   * This is also not the same as being equal according to the `===`` operator.
+   * The `===`` operator (and the `==`` operator as well) treats the number
+   * values `-0`` and `+0`` as equal and treats `Number.NaN`` as not equal to
+   * `NaN`.
+   *
+   * MDN
+   */
+  def is(value1: Any, value2: Any): Boolean = native
```

Comment:
I think the below cases are not covered by `eq` 

* Comparing `Double.NaN` and `Double.NaN`. `Object.is` return `true`, but `==` return `false`. 
* Comparing `+0` and `-0`. `Object.is` return `false`, but  `==` return `true`.

IIUC, `eq` cannot be usable to `Double` since `Double` is not `AnyRef`.

Reply:
The `NaN` and `+0`/`-0` cases are covered by `eq` in Scala.js 1.x; not in 0.6.x.

> IIUC, `eq` cannot be usable to `Double` since `Double` is not `AnyRef`.

You can cast the `Double` to `AnyRef` (or ascribe it to `java.lang.Double`, which is less ugly), but fair enough.